src/consumers/index.js

```diff
@@ -0,0 +1,109 @@
+const config = require('~/src/config')
+const amqplib = require('amqplib')
+const logger = require('~/src/logging').logger(module)
+
+const QueueConsumer = require('./QueueConsumer')
+
+const EVENTS_QUEUE_NAME = config.getEventsQueueName()
+const EVENTS_QUEUE_PREFETCH_COUNT = config.getEventsQueuePrefetchCount()
+const WORK_QUEUE_NAME = config.getWorkQueueName()
+const WORK_QUEUE_PREFETCH_COUNT = config.getWorkQueuePrefetchCount()
+
+const CONSUMER_RECONNECT_TIMEOUT = config.getConsumerReconnectTimeout()
+
+let connectionPromise
+let amqConnection
+
+/**
+ * Creates a connection
+ */
+async function _getConnection () {
+  if (connectionPromise) {
+    return connectionPromise
+  }
+
+  if (!amqConnection) {
+    logger.info(`Attempting to connect to amq ${config.getAmqUrl()}`)
+    connectionPromise = amqplib.connect(config.getAmqUrl())
+      .then((connection) => {
+        amqConnection = connection
+        connectionPromise = null
+        return connection
+      })
+      .catch((err) => {
+        connectionPromise = null
+        amqConnection = null
+        throw new Error('Unable to connect to ActiveMQ', err)
+      })
+
+    return connectionPromise
+  } else {
+    return amqConnection
+  }
+}
+
+async function _startConsumer (queueName, prefetchCount) {
+  try {
+    const connection = await _getConnection()
+    const consumer = new QueueConsumer({
+      connection,
+      queueName,
+      prefetchCount
+    })
+
+    consumer.on('message', async (message) => {
+      try {
+        // TODO: handle messages here
+
+        // acknowledge the message on success
+        await consumer.acknowledgeMessage(message)
+      } catch (err) {
+        logger.error('Error handling message:', err)
+
+        // reject/requeue message
+        await consumer.rejectMessage(message)
+      }
+    })
```

Comment:
It might be a good idea to use round robin queuing here to ensure that we evenly process messages from both `events` and `work`. `drr-fair-queue` is a nice module that could take care of this.